### PR TITLE
Fixes Icon-related typings

### PR DESCRIPTION
**Pull request by dzucconi**
Not sure how these were breaking in master, but that lockfile regeneration apparently did it.

### CODE REVIEW

File: packages/palette/src/svgs/Icon.tsx [hunk 1]
```diff
@@ -1,42 +1,20 @@
...
-} from "styled-system"
+import { position, PositionProps, space, SpaceProps } from "styled-system"
 import { Color } from "../Theme"
 
-// : React.SVGProps<SVGSVGElement>
-
-// tslint:disable-next-line:no-empty-interface
 export interface IconProps
-  extends React.SVGProps<any>,
+  extends React.HTMLAttributes<SVGElement>,
```
Comment (dzucconi) [inline packages/palette/src/svgs/Icon.tsx:7]: This one is what we wanted. The rest is just styled-system clean up.

File: packages/palette/src/svgs/Icon.tsx [hunk 2]
```diff
@@ -1,42 +1,20 @@
...
-    RightProps,
-    LeftProps {
+    PositionProps {
   fill?: Color
   title?: string
 }
 
 /** Wrapper for icons to include space */
-export const Icon = styled.svg.attrs<IconProps>({})`
+export const Icon = styled.svg<IconProps>`
```
Comment (dzucconi) [inline packages/palette/src/svgs/Icon.tsx:15]: I should bulk fix all these `attrs<Type>({})` things

**Comment by damassi** [comment]
LGTM 👍

File: packages/palette/src/svgs/Icon.tsx (hunk 1 above)
Comment (dzucconi) [reply packages/palette/src/svgs/Icon.tsx:15]: (`React.SVGAttributes<SVGSVGElement>` rather... but the issue lies elsewhere)

**Pull request closed without merging**